processing/src/main/java/io/druid/query/join/DimExtractPredicate.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.join;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.druid.query.dimension.DimensionSpec;
+
+import java.util.Objects;
+
+public class DimExtractPredicate implements JoinPredicate
```

Comment:
Hmm, this class simply represents a dimension in join predicates. For example, given a sql ```SELECT count(*) from t1 JOIN t2 ON t1.bar = t2.bar```, ```t1.bar``` is a DimExtractPredicate. Maybe DimensionPredicate is more appropriate.